portal/src/walrus-sites-sw.ts

```diff
@@ -322,31 +336,37 @@ async function resolveAndFetchPage(parsedUrl: Path): Promise<Response> {
  * Fetches a page.
  */
 async function fetchPage(client: SuiClient, objectId: string, path: string): Promise<Response> {
-    const resource = await fetchResource(client, objectId, path);
-    if (resource === null || !resource.blob_id) {
+    const result = await fetchResource(client, objectId, path, new Set<string>);
+    const unsuccessfulResourceFetch = !isResource(result)
+    if (unsuccessfulResourceFetch) {
+        const httpStatus = result as number;
+        return new Response("Unable to fetch the site resource.", { status: result });
```

Comment:
Question: How is this rendered? Should this be using `Response404` to show the error nicely?

Reply:
Right now it doesn't get rendered as an html page, so yeah this is not pretty by the user's perspective.

The reason I didn't use Response404 is because the error status returned by `fetchResource` can be different than this (e.g. LOOP_DETECTED, TOO_MANY_REDIRECTS etc). 

I think it would be nice having another template for `300` errors and `500` errors. 
We can make an issue about this and implement it on another PR.

Reply:
That makes sense. We could maybe rename `Response404` to `ResponseError`, and have each error code and a message passed to it, s.t. both the error code and the message are printed. Wdyt?

Reply:
Sounds good to me!

Reply:
Created this issue: https://github.com/MystenLabs/walrus-sites/issues/122